packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -60,6 +60,22 @@ abstract class FlutterCommand extends Command {
     _usesPubOption = true;
   }
 
+  void addBuildModeFlags() {
+    argParser.addFlag('debug',
```

Comment:
having these be flags means it's easy to end up specifying mutually-exclusive flags.
Maybe `--build-mode=debug` or some such?
But that's non-ergonomic.
If we stick with flags, we should be very elaborate in our handling of the situation where mutually exclusive flags are given at the same time.